chia/rpc/wallet_rpc_api.py

```diff
@@ -1388,6 +1389,16 @@ async def nft_get_nfts(self, request) -> Dict:
             nft_info_list.append(nft_puzzles.get_nft_info_from_puzzle(nft))
         return {"wallet_id": wallet_id, "success": True, "nft_list": nft_info_list}
 
+    async def nft_get_by_did(self, request) -> Dict:
+        did_id: Optional[bytes32] = None
+        if "did_id" in request:
+            did_id = bytes32.from_hexstr(request["did_id"])
+        assert self.service.wallet_state_manager is not None
+        for wallet in self.service.wallet_state_manager.wallets.values():
+            if isinstance(wallet, NFTWallet) and wallet.get_did() == did_id:
+                return {"wallet_id": wallet.wallet_id, "wallet_info": wallet.wallet_info, "success": True}
```

Comment:
I don't think we need to include `wallet_info`. The client doesn't do anything with this data today, and likely won't in the future.

Reply:
I thought you may want to check if a NFT already in a wallet or something else. If you don't need this I can remove it.

Reply:
I appreciate that. For the intended use case, I can call `nft_get_nfts` for the returned `wallet_id`. I think that fits well with what the client needs.

`wallet_info` is really an internal implementation detail that shouldn't be exposed to the RPC callers.